src/Components/LoginForm/index.jsx

```diff
@@ -0,0 +1,152 @@
+import { AuthContext, validateInput } from '../../utils';
+
+import styles from './styles.module.scss';
+
+export const LoginForm = () => {
+
+    const { isUserAuthorized, setAuthStatus } = useContext(AuthContext);
+
+    const [formData, setFormData] = useState({
+        nameFiled: {
+          fieldValue: '',
+          isFieldTouched: false,
+          isFieldCorrect: false,
+          isFieldEmpty: true,
+          invalidCredentialsError: false
+        },
+        passwordField: {
+          fieldValue: '',
+          isFieldTouched: false,
+          isFieldCorrect: false,
+          isFieldEmpty: true,
+          isFieldPassValidation: false
+        },
+        invalidCredentialsError: false
+      }
+    );
+
+    const handleLoginChange = useCallback((value) => {
+      setFormData((prevState) => ({
+        ...prevState,
+        nameFiled: {
+          ...prevState.nameFiled,
+          fieldValue: value
+        },
+        invalidCredentialsError: false
+      }));
+    }, []);
+
+    const handlePasswordChange = (value) => {
+      setFormData((prevState) => ({
+        ...prevState,
+        passwordField: {
+          ...prevState.passwordField,
+          fieldValue: value
+        },
+        invalidCredentialsError: false
+      }));
+
+    };
+
+    const handleNameValidation = (value) => {
+      const {
+        correct,
+        empty,
+        valid
+      } = validateInput(value, 'name');
+
+      setFormData(prevState => ({
+        ...prevState,
+        nameFiled: {
+          ...prevState.nameFiled,
+          fieldValue: value,
+          isFieldTouched: true,
+          isFieldCorrect: correct,
+          isFieldEmpty: empty,
+          isFieldPassValidation: valid
+        }
+      }));
+    };
+
+    const handlePasswordValidation = (value) => {
+      const {
+        correct,
+        empty,
+        valid
+      } = validateInput(value, 'password');
+      setFormData(prevState => ({
+        ...prevState,
+        passwordField: {
+          ...prevState.passwordField,
+          fieldValue: value,
+          isFieldTouched: true,
+          isFieldCorrect: correct,
+          isFieldEmpty: empty,
+          isFieldPassValidation: valid
+        }
+      }));
+    };
+
+    const handleFormSubmit = (e) => {
+      e.preventDefault();
+      if (formData.nameFiled.isFieldCorrect && formData.passwordField.isFieldCorrect) {
+        return setAuthStatus(!isUserAuthorized);
+      }
+      setFormData(prevState => ({
+        ...prevState,
+        passwordField: {
+          ...prevState.passwordField,
+          fieldValue: '',
+          isFieldTouched: false
+        },
+        invalidCredentialsError: true
+      }));
+    };
+
+    const isButtonDisabled = () => {
+      const nameFieldCheck = !formData.nameFiled.isFieldEmpty && formData.nameFiled.isFieldTouched;
+      const passwordFieldCheck = !formData.passwordField.isFieldEmpty && formData.passwordField.isFieldTouched;
```

Comment:
in case you have some doubts around what should your `useInput` hook deal with - please check `formik` for an example. I think this app will require much more simplified approach but anyway, just for an example